internal/client/client.go

```diff
@@ -1,17 +1,202 @@
 )
 
+const (
+	credentialsFileName = "credentials"
+	permifyDirName      = ".permify"
+)
+
+var logger *zap.Logger
+
+type Credentials struct {
+	Endpoint    string `json:"endpoint"`
+	Token       string `json:"token"`
+	CertPath    string `json:"certPath"`
+	CertKeyPath string `json:"certKeyPath"`
+}
+
+func init() {
+	config := zap.NewDevelopmentConfig()
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
+	config.EncoderConfig.EncodeTime = customTimeEncoder
+	logger, _ = config.Build()
+	defer logger.Sync()
+	logger.Info("Client initialized")
+}
+
+func customTimeEncoder(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
+	enc.AppendString(t.Format("2006-01-02 15:04:05.000"))
+}
+
 func New(endpoint string) (*permify.Client, error) {
+	creds, err := retrieveCredentials()
+	if err != nil {
+		if errors.Is(err, errCredentialsNotFound) {
+			// Credentials not found, get from user?
+			if err := getCredentialsFromUser(); err != nil {
+				return nil, fmt.Errorf("failed to create credentials: %v", err)
+			}
+			return New(endpoint)
+		}
+		logger.Error("Failed to retrieve credentials from file", zap.Error(err))
+		return nil, fmt.Errorf("failed to retrieve credentials: %v", err)
+	}
+
+	if err := validateCredentials(creds); err != nil {
+		logger.Warn("Invalid credentials", zap.Any("invalidCredentials", creds), zap.Error(err))
+		return nil, fmt.Errorf("invalid credentials: %v", err)
+	}
+
 	client, err := permify.NewClient(
 		permify.Config{
-			Endpoint: endpoint,
+			Endpoint: creds.Endpoint,
+			// Other necessary configurations
 		},
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
 	)
-	return client, err
+	if err != nil {
+		logger.Error("Failed to create Permify client", zap.Error(err))
+		return nil, fmt.Errorf("failed to create Permify client: %v", err)
+	}
+
+	logger.Info("Permify client created successfully")
+
+	return client, nil
+}
+
+var errCredentialsNotFound = errors.New("credentials file not found")
+
+func retrieveCredentials() (*Credentials, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		logger.Error("Failed to get user home directory", zap.Error(err))
+		return nil, fmt.Errorf("failed to get user home directory: %v", err)
+	}
+
+	credentialsPath := filepath.Join(homeDir, permifyDirName, credentialsFileName)
+
+	fileContent, err := os.ReadFile(credentialsPath)
+	if os.IsNotExist(err) {
+		return nil, errCredentialsNotFound
+	} else if err != nil {
+		logger.Error("Failed to read credentials file", zap.String("file", credentialsPath), zap.Error(err))
+		return nil, fmt.Errorf("failed to read credentials file '%s': %v", credentialsPath, err)
+	}
+
+	var creds Credentials
+	if err := json.Unmarshal(fileContent, &creds); err != nil {
+		logger.Error("Failed to unmarshal credentials", zap.String("file", credentialsPath), zap.Error(err))
+		return nil, fmt.Errorf("failed to unmarshal credentials: %v", err)
+	}
+
+	return &creds, nil
+}
+
+func getCredentialsFromUser() error {
```

Comment:
Whether we need this or no is subjective. as this can be done by user manually.
i'll remove this if you want me to.